### ISSUE TITLE
`dissoc` doesn't work correctly with `TransientRow` (in `after-select`, etc.)

### ISSUE THREAD

**Issue by camsaul** [camsaul/toucan2#105]
I don't know if we should insert some sort of tombstone in this case or what, but `dissoc` doesn't work like we'd expect:

```clj
(derive :models/people.cool :models/people)

(t2/define-after-select :models/people.cool
  [person]
  (assoc person :cool-name (str "Cool " (:name person))))

(t2/select :models/people.cool 1)

(t2/define-after-select ::without-created-at
  [row]
  (println "row:" row) ; NOCOMMIT
  (dissoc row :created_at))

(derive :models/people.cool.without-created-at :models/people.cool)
(derive :models/people.cool.without-created-at ::without-created-at)

(m/prefer-method! #'toucan2.tools.after-select/after-select
                  :models/people.cool
                  ::without-created-at)

(t2/select-one :models/people.cool.without-created-at 1)
;; =>
(toucan2.instance/instance
 :models/people.cool.without-created-at
 {:name "Cam",
  :cool-name "Cool Cam",
  :id 1,
  :created_at #object[java.time.OffsetDateTime 0x6e094282 "2020-04-21T23:56Z"]})

```

**Comment by calherries** [comment]
I ran into this issue on this PR https://github.com/metabase/metabase/pull/31282. `dissoc` doesn't work in an after-select method.

**Comment by camsaul** [comment]
Fixed in `1.0.529`
